capsules/extra/src/eeprom.rs

```diff
@@ -0,0 +1,259 @@
+//! Driver for the AT24C32/64 EEPROM memory. Built on top of the I2C interface.
+//! Provides interface for the NonvolatileToPages driver.
+//!
+//! Datasheet:
+//! <https://ww1.microchip.com/downloads/en/devicedoc/doc0336.pdf>
+//!
+//! > The AT24C32/64 provides 32,768/65,536 bits of serial electrically erasable and programmable
+//! > read only memory (EEPROM) organized as 4096/8192 words of 8 bits each. The device’s cascadable
+//! > feature allows up to 8 devices to share a common 2- wire bus. The device is optimized for use
+//! > in many industrial and commercial applica- tions where low power and low voltage operation are
+//! > essential. The AT24C32/64 is available in space saving 8-pin JEDEC PDIP, 8-pin JEDEC SOIC,
+//! > 8-pin EIAJ SOIC, and 8-pin TSSOP (AT24C64) packages and is accessed via a 2-wire serial
+//! > interface.
+//!
+//! Usage
+//! -----
+//!
+//! ```rust
+//! let i2cmux = I2CMuxComponent::new(i2c0, None).finalize(components::i2c_mux_component_static!());
+//!
+//! let eeprom_buffer = static_init!([u8; 34], [0; 34]);
+//!
+//! let eeprom_i2c_device = static_init!(I2CDevice, I2CDevice::new(i2cmux, 0x50));
```

Comment:
`0x50` should be a named constant in this file. Like: https://github.com/tock/tock/blob/master/capsules/extra/src/bmp280.rs#L21

We have not been good about doing this generally though, and many drivers do exactly what you did here.

Reply:
`0x50` just specifies the i2c address so it's more depends on that how user connects it's eeprom, that is why I refrained from that

Reply:
What is more concerning to me is buffer initialisation - it should always have size `PAGE_SIZE + 2` so we can define it inside the `new()`. Is there a way to initialise a buffer without referencing to kernel?